File: Data Structures & Algorithms/length-of-last-word/submission-0.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        sList = s.split(" ");
        while sList[-1] == "":
            sList.remove(sList[-1])
        count = 0
        for char in sList[-1]:
            count += 1
        
        return count

File: Data Structures & Algorithms/kth-distinct-string-in-an-array/submission-0.java
class Solution {
    public String kthDistinct(String[] arr, int k) {
        Set<String> orderedSet = new LinkedHashSet<>();
        ArrayList<String> duplicates = new ArrayList<>();
        for (String s : arr){
            if (orderedSet.contains(s)){
                duplicates.add(s);
            }
            orderedSet.add(s);
        }
        for (String s : duplicates){
            orderedSet.remove(s);
        }
        if (orderedSet.size() < k){
            return "";
        }
        int count = 1;
        for (String s : orderedSet){
            if (count == k){
                return s;
            }
            count++;
        }
        return "";
    }
}